pkgs/by-name/gp/gpustat/package.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, fontconfig
+, libGL
+, libX11
+, libXcursor
+, libXi
+, libXrandr
+, cmake
+, libxkbcommon
+, wayland
+, makeWrapper
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "gpustat";
+  version = "0.1.4";
+
+  src = fetchFromGitHub {
+    owner = "arduano";
+    repo = "gpustat";
+    rev = "v${version}";
+    hash = "sha256-bgw9KrSiswfpKHn+qP1IiYgIuRzje+Ql/Mr0VqHvrbQ=";
+  };
+
+  cargoHash = "sha256-sUxuiOjBIn/5jckezX6S6TQtoiqjHnlOn8fRdaVRwYM=";
+
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+    pkg-config
+  ];
+
+  buildInputs = [
+    fontconfig
+    libGL
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    libxkbcommon
+    wayland
+  ];
+
+  postInstall = ''
+    mkdir -p $out/share/applications $out/share/pixmaps
+
+    cp assets/gpustat.desktop $out/share/applications
+    cp assets/icon_* $out/share/pixmaps
```

Comment:
It is recommended that you change the name of the installed icon to a name specific to the application. Of course, the icon in the .desktop file also needs to be changed.